src/parent/reducers/reducers.js

```diff
@@ -1,12 +1,72 @@
 import { combineReducers } from 'redux';
+import reduceReducers from 'reduce-reducers';
+
+import { PARENT as ACTION_TYPES } from '../../shared/constants/actionTypes';
 
 import childWindows from './childWindows';
+import closedWindows from './closedWindows';
 import dragOut from './dragOut';
 
-const rootReducer = combineReducers({
-    childWindows,
-    dragOut
-});
+const rootReducer = reduceReducers(
+    combineReducers({
+        childWindows,
+        closedWindows,
+        dragOut
+    }),
+    (state, action) => {
+        switch (action.type) {
+        case ACTION_TYPES.REOPEN: {
+            const newClosedWindow = Object.assign({}, state.closedWindows[action.windowName], {});
+
+            delete newClosedWindow.date;
+
+            const newClosedWindows = Object.assign({}, state.closedWindows);
+            delete newClosedWindows[action.windowName];
+
+            return {
+                childWindows: Object.assign({}, state.childWindows, {
+                    [action.windowName]: newClosedWindow
+                }),
+                closedWindows: newClosedWindows,
+                dragOut: state.dragOut
+            };
+        }
+
+        case ACTION_TYPES.CLOSE: {
+            const newClosedWindow = Object.assign({}, state.childWindows[action.windowName], {
+                date: action.date
+            });
+
+            const newChildWindows = Object.assign({}, state.childWindows);
+
+            delete newChildWindows[action.windowName];
+
+            const numberOfFavourites = state.childWindows[action.windowName].favourites.codes.length;
```

Comment:
I'd probably do the following:

`const closingWindowHasFavourites = state.childWindows[action.windowName].favourites.codes.length > 0`

As I think it's slightly more clear what's happening. Could even do this check earlier and return early - I think it's fine as-is though.